flink-runtime/src/main/java/org/apache/flink/runtime/externalresource/ExternalResourceUtils.java

```diff
@@ -118,6 +119,35 @@ private ExternalResourceUtils() {
 		return externalResourceConfigs;
 	}
 
+	/**
+	 * Get the map of resource name and amount of all of enabled external resources.
+	 */
+	private static Map<String, Long> getExternalResourceAmountMap(Configuration config) {
+		final Set<String> resourceSet = getExternalResourceSet(config);
+		LOG.info("Enabled external resources: {}", resourceSet);
+
+		if (resourceSet.isEmpty()) {
+			return Collections.emptyMap();
+		}
+
+		final Map<String, Long> externalResourceAmountMap = new HashMap<>();
+		for (String resourceName: resourceSet) {
+			final ConfigOption<Long> amountOption =
+				key(ExternalResourceOptions.keyWithResourceNameAndSuffix(resourceName, ExternalResourceOptions.EXTERNAL_RESOURCE_AMOUNT_SUFFIX))
+					.longType()
+					.noDefaultValue();
+			final Long amount = config.getLong(amountOption, 0);
+			if (amount <= 0) {
+				LOG.warn("The amount of the {} should be configured with a positive value.", resourceName);
+				break;
```

Comment:
Same here. Let's separate the two cases of not having amount and having illegal amount. Also, log that the amount is ignored.